src/Azure/Shared/Storage/AzureTableDataManager.cs

```diff
@@ -216,6 +218,45 @@ public async Task<string> UpsertTableEntryAsync(T data)
             }
         }
 
+        /// <summary>
+        /// Inserts a data entry in the Azure table: creates a new one if does not exists
+        /// </summary>
+        /// <param name="data">Data to be inserted or replaced in the table.</param>
+        /// <returns>Value promise with new Etag for this data entry after completing this storage operation.</returns>
+        public async Task<(bool isSuccess, string eTag)> InsertTableEntryAsync(T data)
+        {
+            const string operation = "InsertTableEntry";
+            var startTime = DateTime.UtcNow;
+            if (Logger.IsEnabled(LogLevel.Trace)) Logger.Trace("{0} entry {1} into table {2}", operation, data, TableName);
+
+            try
+            {
+                try
+                {
+                    // WAS:
+                    // svc.AttachTo(TableName, data, null);
+                    // svc.UpdateObject(data);
+                    // SaveChangesOptions.ReplaceOnUpdate,
+                    var opResult = await tableReference.ExecuteAsync(TableOperation.Insert(data));
+                    return (true, opResult.Etag);
+                }
+                catch (StorageException storageException) when (storageException?.RequestInformation?.HttpStatusCode == (int)HttpStatusCode.Conflict)
+                {
+                    return (false, null);
+                }
+                catch (Exception exc)
+                {
+                    Logger.Warn((int)Utilities.ErrorCode.AzureTable_06,
```

Comment:
This is copy-pasted from above. Can we avoid code duplication like that with a helper method instead? There's a number of similar cases in this code file.

Reply:
BTW, I don't understand why we same "Intermediate" here.